src/Commands/Generators/UserModelGenerator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CodeIgniter\Shield\Commands\Generators;
+
+use CodeIgniter\CLI\BaseCommand;
+use CodeIgniter\CLI\CLI;
+use CodeIgniter\CLI\GeneratorTrait;
+
+/**
+ * Generates a custom user model file.
+ */
+class UserModelGenerator extends BaseCommand
+{
+    use GeneratorTrait;
+
+    /**
+     * @var string
+     */
+    protected $group = 'Shield';
+
+    /**
+     * @var string
+     */
+    protected $name = 'shield:model';
+
+    /**
+     * @var string
+     */
+    protected $description = 'Generate a new UserModel file.';
+
+    /**
+     * @var string
+     */
+    protected $usage = 'shield:model [<name>] [options]';
+
+    /**
+     * @var array<string, string>
+     */
+    protected $arguments = [
+        'name' => 'The model class name. If not provided, this will default to `UserModel`.',
+    ];
+
+    /**
+     * @var array<string, string>
+     */
+    protected $options = [
+        '--namespace' => 'Set root namespace. Default: "APP_NAMESPACE".',
+        '--suffix'    => 'Append the component title to the class name (e.g. User => UserModel).',
+        '--force'     => 'Force overwrite existing file.',
+    ];
+
+    /**
+     * Actually execute the command.
+     */
+    public function run(array $params): void
+    {
+        $this->component = 'Model';
+        $this->directory = 'Models';
+        $this->template  = 'usermodel.tpl.php';
+
+        $this->classNameLang = 'CLI.generator.className.model';
+        $this->setHasClassName(false);
+
+        $class = $params[0] ?? CLI::getSegment(2) ?? 'UserModel';
+
+        if (in_array(strtolower($class), ['shielduser', 'shieldusermodel'], true)) {
```

Comment:
```suggestion
        if (in_array(strtolower($class), ['shielduser', 'shieldusermodel', 'shielduser_model', 'shield_user_model', 'shield_user', 'shield_user_'], true)) {
```
Although nothing should be chosen as the **class name**. Just for note. And probably a few more that I don't know.

Reply:
Why do we need `shielduser`? Cannot generate class `ShieldUser`?

Reply:
@datamweb If you are okay, please merge this.

Reply:
@paulbalandan thank you.
 Really, in many cases, your deep understanding of the code amazes me.